blog/2017-08-10-RealTimePlanningUsingKieServerPart1.adoc

```diff
@@ -0,0 +1,149 @@
+      </plugin>
+    </plugins>
+  </build>
+----
+
+. Create a container (server unit of execution) wrapping the project.
+This example uses Java KIE Server client, which requires the following dependencies in your client application:
++
+[source,xml,options="nowrap"]
+----
+  <dependency>
+    <groupId>org.kie.server</groupId>
+    <artifactId>kie-server-api</artifactId>
+    <version>7.2.0.Final</version>
+  </dependency>
+  <dependency>
+    <groupId>org.kie.server</groupId>
+    <artifactId>kie-server-client</artifactId>
+    <version>7.2.0.Final</version>
+  </dependency>
+----
+Client and container creation:
++
+[source,java,options="nowrap"]
+----
+  KieServicesConfiguration kieServicesConfiguration = new KieServicesConfigurationImpl(SERVER_URL, USERNAME, PASSWORD, CLIENT_TIMEOUT);
+  KieServicesClient kieServicesClient = KieServicesFactory.newKieServicesClient(kieServicesConfiguration)
+
+  // ReleaseId corresponds to groupId:artifactId:version (GAV) of the project installed in the first step
+  ReleaseId releaseId = new ReleaseId("org.optaplanner", "kie-server-example", "1.0.0-SNAPSHOT");
+  KieContainerResource containerResource = new KieContainerResource(CONTAINER_ID, releaseId);
+  kieServicesClient.createContainer(CONTAINER_ID, containerResource);
+----
+
+. Create a solver within the container.
+A single container can hold multiple solvers, each of them having a unique identifier.
+This is useful if you want to try out different solver configurations.
++
+[source,java,options="nowrap"]
+----
+SolverServicesClient solverClient = kieServicesClient.getServicesClient(SolverServicesClient.class);
+solverClient.createSolver(CONTAINER_ID, SOLVER_ID, SOLVER_CONFIG_XML_PATH);
+----
+
+. Submit your optimization problem
++
+[source,java,options="nowrap"]
+----
+// planningProblem parameter is an instance of your PlanningSolution class
+solverClient.solvePlanningProblem(CONTAINER_ID, SOLVER_ID, planningProblem);
+----
+
+. Check the solver state.
+At this point of time the best solution notifications are not supported yet on the client side,
+you need to periodically check the solver status/best solution.
++
+[source,java,options="nowrap"]
+----
+  SolverInstance solver = solverClient.getSolver(CONTAINER_ID, SOLVER_ID);
+  if (solver.getStatus() == SolverInstance.SolverStatus.SOLVING) {
+    // continue
+  } else {
+    Solution_ solution = (Solution_) solverClient.getSolverWithBestSolution(CONTAINER_ID, SOLVER_ID).getBestSolution()
+    // process the solution
+  }
+----
+
+. Once you are done with the optimization, terminate the solver.
+If you don't plan to reuse the solver anymore, dispose it to free the resources on the server.
++
+[source,java,options="nowrap"]
+----
+  solverClient.terminateSolverEarly(CONTAINER_ID, SOLVER_ID);
+  // or
+  solverClient.disposeSolver(CONTAINER_ID, SOLVER_ID);
+----
+
+== Where do I start?
+Download KIE (Execution) Server http://www.optaplanner.org/download/download.html[here].
```

Comment:
remove the word "here" and wrap the link around the action, so around "Download KIE (Execution) Server"